Subject: [PATCH] jdk21u_android
---
Index: src/hotspot/share/utilities/elfFile.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hotspot/share/utilities/elfFile.hpp b/src/hotspot/share/utilities/elfFile.hpp
--- a/src/hotspot/share/utilities/elfFile.hpp	(revision 374eaf89e67d07479b1cc744b7f190b1836dbc84)
+++ b/src/hotspot/share/utilities/elfFile.hpp	(revision ccd7f774bc9a3e790b9025d32e4df145999cbcca)
@@ -46,7 +46,7 @@
 typedef Elf64_Phdr      Elf_Phdr;
 typedef Elf64_Sym       Elf_Sym;
 
-#if !defined(_ALLBSD_SOURCE) || defined(__APPLE__)
+#if (!defined(__ANDROID__) && !defined(_ALLBSD_SOURCE)) || defined(__APPLE__)
 #define ELF_ST_TYPE ELF64_ST_TYPE
 #endif
 
